src/PrestaShopBundle/Resources/config/services/bundle/serializer.yml

```diff
@@ -0,0 +1,17 @@
+services:
+  _defaults:
+    public: true
+
+  # Snake case normalizer definition
+  prestashop.bundle.snake_case_normalizer.object:
+    class: Symfony\Component\Serializer\Normalizer\ObjectNormalizer
+    arguments:
+      - ~
+      - '@serializer.name_converter.camel_case_to_snake_case'
+
+  # Snake case serializer definition
+  prestashop.bundle.snake_case_serializer_json:
```

Comment:
Maybe it would be nice to have tests for it?

Reply:
It is just a specifically configured symfony serializer. Is it really necessary?

Reply:
Hmmmm, not sure, maybe not if it's covered by symfony itself.

Reply:
Im not sure its worth introducing snake case serializer at all.
Symfony controllers already have $this->json() method for json response. As JSON doesn't really have any strict convention, it seems like an overhead to me.

Reply:
or maybe it does make sense after this - https://github.com/PrestaShop/PrestaShop/issues/15977#issuecomment-542780827